From 32bad8ed982d70e88cebd704abb601d8a6246af9 Mon Sep 17 00:00:00 2001
From: Jerin Jacob <jerinj@marvell.com>
Date: Thu, 25 Oct 2018 13:17:04 +0530
Subject: [PATCH 0367/1051] soc: octeontx2: Update workaround for HW NIX errata
 35057

HW errata 35057 updated to set NIX_AF_CFG[FORCE_PSE_CLK_EN] = 1
as well.Update the bitmask accordingly.

Errata 35057
   NIX may corrupt internal state when conditional clocks turn off.

Signed-off-by: Jerin Jacob <jerinj@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_nix.c b/drivers/soc/marvell/octeontx2/rvu_nix.c
index 457917a61576..e0624f0682af 100644
--- a/drivers/soc/marvell/octeontx2/rvu_nix.c
+++ b/drivers/soc/marvell/octeontx2/rvu_nix.c
@@ -2664,7 +2664,7 @@ int rvu_nix_init(struct rvu *rvu)
 	 */
 	if (is_rvu_9xxx_A0(rvu))
 		rvu_write64(rvu, blkaddr, NIX_AF_CFG,
-			    rvu_read64(rvu, blkaddr, NIX_AF_CFG) | 0x1EULL);
+			    rvu_read64(rvu, blkaddr, NIX_AF_CFG) | 0x5EULL);
 
 	/* Calibrate X2P bus to check if CGX/LBK links are fine */
 	err = nix_calibrate_x2p(rvu, blkaddr);
-- 
2.17.1

